projects/components/src/stepper/stepper.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  EventEmitter,
+  Input,
+  Output,
+  QueryList
+} from '@angular/core';
+import { MatStepper } from '@angular/material/stepper';
+import { queryListAndChanges$ } from '@hypertrace/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ButtonRole, ButtonStyle } from '../button/button';
+import { IconSize } from '../icon/icon-size';
+import { StepperTabComponent } from './stepper-tab.component';
+
+@Component({
+  selector: 'ht-stepper',
+  template: `
+    <div class="stepper" *ngIf="this.steps$ | async as steps">
+      <mat-stepper [orientation]="this.orientation" [linear]="this.isLinear" #stepper>
+        <ng-container *ngFor="let step of steps; index as i">
+          <mat-step [completed]="step.completed" [stepControl]="step.stepControl" [optional]="step.optional">
+            <ng-template matStepLabel>
+              <ht-label class="header-label" [label]="step.label"></ht-label>
+            </ng-template>
+            <ng-container *ngTemplateOutlet="step.content"></ng-container>
+            <ng-template matStepperIcon="edit">
+              <ht-icon [icon]="step.icon" [size]="'${IconSize.ExtraSmall}'"> </ht-icon>
+            </ng-template>
+          </mat-step>
+        </ng-container>
+      </mat-stepper>
+      <div class="action-buttons">
+        <ht-button label="Cancel" (click)="onClickCancel()" [role]="'${ButtonRole.Tertiary}'"></ht-button>
+        <div class="back-next">
+          <ht-button
+            *ngIf="stepper.selectedIndex !== 0"
+            [display]="'${ButtonStyle.Outlined}'"
+            label="Back"
+            (click)="onClickBack(stepper)"
+          ></ht-button>
+          <ht-button
+            *ngIf="stepper.selectedIndex !== steps.length - 1"
+            label="Next"
+            (click)="onClickNext(stepper)"
+            [disabled]="
+              this.isNextDisabled
```

Comment:
nit: Exist some way to improve this? probably using some boolean variables or something similar.

Reply:
Yes, we can make a directive for the same. Added a TODO.

Reply:
You're just passing boolean variables from here and just doing `&&` in the function which is not making any difference.
I think we can
1. Either just pass stepper from here to the function (we don't event need `isLinear` to be passed from here) or
2. do those `&&` operations directly here

Reply:
Updated. Implemented the first idea of passing `stepper`.